src/wandb_osh/syncer.py

```diff
@@ -44,44 +66,42 @@ def sync(self, dir: PathLike) -> None:
         """
         sync_dir(dir, options=self.wandb_options, timeout=self._timeout)
 
-    def loop(self) -> None:
+    def dir_watcher(self) -> None:
         """Read command files and trigger syncing"""
         logger.info(
             "wandb-osh v%s, starting to watch %s", __version__, self.command_dir
         )
         while True:
             start_time = time.time()
             self.command_dir.mkdir(parents=True, exist_ok=True)
-            command_files = []
-            targets = []
             for command_file in self.command_dir.glob("*.command"):
                 target = Path(command_file.read_text())
-                command_files.append(command_file)
                 if not target.is_dir():
                     logger.error(
                         "Command file %s points to non-existing directory %s",
                         command_file,
                         target,
                     )
                     continue
-                targets.append(target)
-            for target in set(targets):
-                logger.info("Syncing %s...", target)
-                try:
-                    self.sync(target)
-                except subprocess.TimeoutExpired:
-                    # try again later
-                    logger.warning("Syncing %s timed out. Trying later.", target)
-                    from wandb_osh.hooks import TriggerWandbSyncHook
-
-                    TriggerWandbSyncHook(self.command_dir)(target)
-            time.sleep(0.25)
-            for cf in command_files:
+                self.target_queue.put((command_file, target))
```

Comment:
In the previous version, we were making sure that we don't have the same synchronization target more than once in order to avoid one process outcrowding the others (as you reported in #83 , that wasn't enough though). I think we should still do this, i.e., we check if the `target` is already in the queue (if it is, we simply remove the `command_file` right here and don't add anything to the queue).